### PR TITLE
Adds LossMinimization folder and two examples

### PR DESCRIPTION
**Pull request by sandreza**
This is meant to be a starting point for a discussion on adding tutorials on how to use calibrate emulate sample with simple examples to build intuition.

In particular PR adds a simple example showing how to use EKI to minimize two loss functions
1. L(x,y) = (x-1)^2 + (y+1)^2
2. L(x,y) = (x^2-1)^2 + (y+1)^2

This is accomplished by taking the forward map to just be the loss function. The former has a minimum at (x,y) = (1,-1) and the latter has two minima at (x,y) = (+1,-1) and (x,y) = (-1,-1). By switching the random seed on the latter, one can converge to either minimum.  Eventually, such simple examples could be included in the docs

### CODE REVIEW

**Comment by odunbar** [comment]
Thanks for starting this! I'd be keen to chat more about some tutorial-type examples. At present we only have the `EnsembleKalmanProcesses/runtests` (`EKP/runtests`) for the ensemble methods (in isolation).

**Comment by sandreza** [comment]
"we can relegate the runtests to more unit-style function testing"
I agree with this!

File: examples/LossMinimization/loss_minimization.jl [hunk 1]
```diff
@@ -0,0 +1,117 @@
+using Distributions
+using LinearAlgebra
+using Random
+using Test
```
Comment (odunbar) [inline examples/LossMinimization/loss_minimization.jl:4]: Test is not used

File: examples/LossMinimization/loss_minimization.jl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+using Plots
+using CalibrateEmulateSample.EKP
```
Comment (odunbar) [inline examples/LossMinimization/loss_minimization.jl:6]: Renamed to `CalibrateEmulateSample.EnsembleKalmanProcesses` in newest master

File: examples/LossMinimization/loss_minimization.jl [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+prior_mean = reshape(get_mean(prior),:)
+prior_cov = get_cov(prior)
+
+# Calibrate
+N_ens = 50  # number of ensemble members
+N_iter = 20 # number of EKI iterations
+initial_ensemble = EKP.construct_initial_ensemble(prior, N_ens;
+                                                rng_seed=rng_seed)
+
+ekiobj = EKP.EKObj(initial_ensemble,
```
Comment (odunbar) [inline examples/LossMinimization/loss_minimization.jl:44]: `EKP` -> `EnsembleKalmanProcesses`
`EKObj` -> `EnsembleKalmanProcess`

File: examples/LossMinimization/loss_minimization.jl (hunk 1 above)
Comment (sandreza) [reply examples/LossMinimization/loss_minimization.jl]: removed

File: examples/LossMinimization/loss_minimization.jl (hunk 2 above)
Comment (sandreza) [reply examples/LossMinimization/loss_minimization.jl]: and renamed!

File: examples/LossMinimization/loss_minimization.jl (hunk 3 above)
Comment (sandreza) [reply examples/LossMinimization/loss_minimization.jl]: also renamed

**Comment by odunbar** [comment]
bors try

**Review by odunbar — APPROVED** [review]

**Pull request merged**